src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -88,16 +87,24 @@ class TooltipRenderedOnPageBody extends React.PureComponent {
 
         // Reset the tooltip text width to 0 so that we can measure it again.
         // eslint-disable-next-line react/no-did-update-set-state
-        this.setState({tooltipContentWidth: 0}, this.updateTooltipTextWidth);
+        this.setState({tooltipContentWidth: 0}, this.updateTooltipContentWidth);
     }
 
-    updateTooltipTextWidth() {
-        if (!this.textRef) {
+    /**
+     * Updates the tooltipContentWidth state with the width of the tooltip's inner content.
+     * Can be called in onLayout, or if you are using text just call it after the first render
+     * to get the text's width.
+     *
+     * @param {Object} [event]
+     */
+    updateTooltipContentWidth(event) {
```

Comment:
This looks more complicated then it should, just use a `width` param. if `textRef` is set set width using `this.textRef.offsetWidth` otherwise set with with passed `width` (which we pass on the view layout callback)